### DISCUSSION TITLE
DAX Studio - error while mapping/checking aggregation "missing CountRows mapping"

### DISCUSSION THREAD

**Discussion by ImranDJCS** [DaxStudio/DaxStudio#603] [General]
Hi there,
I'm fairly new with DAX Studio. While creating and verifying my Power BI aggregations if they are really hitting aggregate tables. I came across this error reported in DAX Studio "reason: missing CountRows mapping".
Can anyone help with this error? What does it mean? 
Please note, I also created counts column in my Agg table but it did not fix the issue.
Thank you

**Comment by ImranDJCS** [comment]
![image](https://user-images.githubusercontent.com/83199486/116030594-39859a00-a69f-11eb-98f0-f6783a69c532.png)

**Comment by dgosbell** [comment]
This means that your query is using a measure that is referencing a CountRows of your fact table, but you do not have a column in your aggregate table that is marked as being a count of the rows in the underlying fact table. You need to create a column like this in your aggregate table/view and mark it as such when using the Manage Aggregations feature in Power BI as per the screenshot below:

![image](https://user-images.githubusercontent.com/8241830/116030464-e875a600-a69e-11eb-946c-24788dfadfe6.png)

**Reply by ImranDJCS** [reply]
Excellent! Really appreciate.
I'm not using a measure that is referencing CountRows. But your advice worked changing to "Count table rows" fixed one of the two queries it is generating.
The other query error message as follows.
![image](https://user-images.githubusercontent.com/83199486/116032560-45735b00-a6a3-11eb-9ee7-a10feee29aed.png)

Appreciate your help.
Thank you

**Reply by dgosbell** [reply]
> I'm not using a measure that is referencing CountRows.

So it may not be an explicit countrows, it could be something like an average which is basically a sum divided by a count of rows.

The data you see in the Details section is the raw message returned from the Microsoft Tabular engine and there is actually not a lot of documentation about the different messages that could be returned. At a guess it sounds like the SFR-9501 does not have an aggregate defined for it.

**Reply by ImranDJCS** [reply]
Thank you dgosbell for pushing me through it. Really appreciate.
I was able to fix and and its working.
Cheers!

**Reply by ImranDJCS** [reply]
Is it the right forum to ask such questions?
Or advise if different. 
Thank you

**Reply by dgosbell** [reply]
This is forum is probably one of the better places to ask questions like this since DAX Studio is one of the few tools to expose these events in an easy to consume way. 

Do you mind if I ask how you fixed the issue? Was my guess correct that you just needed to add an aggregate table for SFR-9501?

**Comment by ImranDJCS** [comment]
Yes! Certainly your advice led me to the explicit counts, which were causing the issue and were not mapping with my Agg table in my composite model.
